lib/puppet/file_serving/http_metadata.rb

```diff
@@ -20,7 +20,21 @@ def initialize(http_response, path = '/dev/null')
       checksum = checksum.unpack("m0").first.unpack("H*").first
       @checksums[:md5] = "{md5}#{checksum}"
     end
+    # Add support for artifactory headers.
+    # This adds support for all three algorithms, sha256, sha1, and md5.
+    if checksum = http_response['X-Checksum-Md5']
```

Comment:
Why doesn't it look right? The file resource type can specify which checksum to use. However, this metadata endpoint has no knowledge of what's expected from the file type, so just appends everything.